controllers/common.go

```diff
@@ -105,15 +106,117 @@ func httpContext(
 	case *v1alpha1.PortalConfig:
 		err = get(o.Spec.Context)
 	}
+
 	if err != nil {
 		return environmet.Env{}, nil, err
 	}
+
 	return e, client.SetContext(ctx, client.Context{
 		Env: e,
 		Log: log,
 	}), nil
 }
 
+func updateOperatorContextStatus(
+	ctx context.Context,
+	rClient runtimeClient.Client,
+	e environmet.Env,
+	object runtimeClient.Object,
+	log logr.Logger,
+	ctxRef *model.Target,
+) error {
+	target := model.Target{
+		Name:      object.GetName(),
+		Namespace: object.GetNamespace(),
+	}
+
+	// Remove link from other operator context, if any,
+	// as we do not know if object was referencing to different context previously
+	var opCtxList v1alpha1.OperatorContextList
+
+	if err := rClient.List(ctx, &opCtxList); err != nil {
+		return err
+	}
+
+	switch object.(type) {
+	case *v1alpha1.ApiDefinition:
+		for _, opctx := range opCtxList.Items {
+			opctx.Status.RemoveLinkedAPIDefinition(target, log)
+
+			err := rClient.Status().Update(ctx, &opctx)
+			if err != nil {
+				log.Error(err, "Failed to remove link of APIDefintion from operator context", "operatorContext", opctx.Name, "apidefinition", target.Name)
```

Comment:
@asoorm Spoted the same issue.
I think we need to roll back all the operations performed before in case of error instead of returning directly. Otherwise it will leave the components in inconsistent state.

Didn't want to handle everything in single PR. But I could change it for now. Definitely need to be handled properly.